synapse/handlers/message.py

```diff
@@ -98,10 +133,19 @@ def get_state_events(self, user_id, room_id, is_guest=False):
         )
 
         if membership == Membership.JOIN:
-            room_state = yield self.state.get_current_state(room_id)
+            if at_event:
+                room_state = yield self.store.get_state_for_events(
```

Comment:
do we not need to check that the user has permission to view the room at this point?